elastic/kibana
# [Console] Set correct initial value in monaco editor

Closes https://github.com/elastic/kibana/issues/177332

## Summary

This PR makes the new Monaco editor for Console to have a correct initial value depending on the `initialTextValue` prop or the `load_from` url query.

### Testing

**Testing with `load_from` query parameter:**

1. Create a `config/kibana.dev.yml` file (if one doesn't exist already) and add the line: `console.dev.enableMonaco: true`
2. Start Es and Kibana
3. Navigate to `/dev_tools#/console?load_from=data:text/plain,OIUQKgBA+gzgpgQwE4GMAWAoA3gIgI4CucSAnjgFy4C2CALulAgDZMVYC+nQA`. This should prepolutate Console with the following request:
```
GET _search
{"query":{"match_all":{}}}
```

Note: The `load_from` functionality should also support loading from a URL (e.g. `app/dev_tools#/console?load_from=https://www.elastic.co/guide/en/elasticsearch/reference/current/snippets/2935.console`) but this will be addressed in a separate PR.

**Testing with `initialTextValue` prop:**
1. Create a `config/kibana.dev.yml` file (if one doesn't exist already) and add the line: `console.dev.enableMonaco: true`
2. Change line 77 in `src/plugins/console/public/application/containers/editor/editor.tsx` to `<MonacoEditor initialTextValue={<addSomeTextHere>} />`
4. Start Es and Kibana
5. Open Console and verify that it contains the text that you provided in the prop.

**Testing with `DEFAULT_INPUT_VALUE` constant:**
1. Create a `config/kibana.dev.yml` file (if one doesn't exist already) and add the line: `console.dev.enableMonaco: true`
2. Change line 77 in `src/plugins/console/public/application/containers/editor/editor.tsx` to `<MonacoEditor initialTextValue={null} />`
3. Start Es and Kibana
5. Open Console and verify that it contains the default text:
```
# Click the Variables button, above, to create your own variables.
GET ${exampleVariable1} // _search
{
  "query": {
    "${exampleVariable2}": {} // match_all
  }
}
```
